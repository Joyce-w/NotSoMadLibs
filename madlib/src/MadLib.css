.MadLib-values{
    display: flex;
    width: 50%;
    margin: auto;
}