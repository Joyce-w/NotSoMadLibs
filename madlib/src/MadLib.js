import React, { useEffect, useState } from "react"
import axios from "axios"
import MadLibForm from "./MadLibForm"
import useAxios from "./hooks/getData"
import "./MadLib.css"

const MadLib = () => {

    const [data, updateData, clearData] = useAxios();

    const [title, setTitle] = useState();
    const [blanks, setBlanks] = useState([]);
    const [answers, setAnswers] = useState();
    const [values, setValues] = useState();

    const [url, setUrl] = useState('http://madlibz.herokuapp.com/api/random?maxlength=11')


    useEffect(() => {
        async function getMadLib() {
            const res = await axios.get(url)
            console.log(res.data)

            updateData(res.data)

            setTitle(res.data.title)
            setBlanks(res.data.blanks)
            setValues(res.data.value)
        }
        getMadLib()
    },[])

    //handle form return data
    const formData = (res) => {
        setAnswers(Object.values(res))
    }

    return (
        <div className="MadLib">
            <h1>MadLib</h1>
            {answers ? null : 
            <MadLibForm title={title} blanks={blanks} formData={formData} />
            }

            <div className="MadLib-values">
                {answers ? answers.map((val, idx) => {
                    return values[idx] + val;
                })
                : ""}
            </div>
        </div>
    )
}

export default MadLib;