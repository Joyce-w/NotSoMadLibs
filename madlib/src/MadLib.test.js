import { render } from '@testing-library/react';
import MadLib from './MadLib';


//smoke test
test('renders learn react link', () => {
  render(<MadLib />);
});


//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<MadLib/>)
  expect(asFragment()).toMatchSnapshot();
})

//specialized testing
test("Test for header", () => {
    const { getByText } = render(<MadLib />)
    
    //get h1 text
    let h1 = getByText("MadLib")
    expect(h1).toBeInTheDocument();
})