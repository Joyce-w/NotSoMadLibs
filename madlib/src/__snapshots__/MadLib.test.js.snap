// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MadLib"
  >
    <h1>
      MadLib
    </h1>
    <div
      class="MadLibForm"
    >
      <form
        class="MadLibForm-form"
      >
        <h1 />
        <button>
          Make some mad libs
        </button>
      </form>
    </div>
    <div
      class="MadLib-values"
    />
  </div>
</DocumentFragment>
`;
